c/include/libsbp/sbp.h

```diff
@@ -87,6 +87,8 @@ void sbp_clear_callbacks(sbp_state_t* s);
 void sbp_state_init(sbp_state_t *s);
 void sbp_state_set_io_context(sbp_state_t *s, void* context);
 s8 sbp_process(sbp_state_t *s, s32 (*read)(u8 *buff, u32 n, void* context));
+s8 sbp_process_generic(sbp_state_t *s, s32 (*read)(u8 *buff, u32 n, void *context),
+    s8 (*sbp_process_payload) (sbp_state_t *s, u16 sender_id, u16 msg_type, u8 msg_len, u8 payload[]));
```

Comment:
I'd be hesitant to have the generic callback take in `sbp_state_t *` directly, the user's code shouldn't be using that. Instead maybe have a `void *context` as an arg to the generic callback and pass it into `sbp_process_generic` as well?

Reply:
Agreed, I think a context was better and more flexible.  For the legacy case, the context is the head of the linked list.